File: server/Models/Device.cs
namespace Server.Models;

public class Device
{
    public string Id { get; set; }
    public string Identifier { get; set; }
    public string Description { get; set; }
    public string Manufacturer { get; set; }
    public string Url { get; set; }
    public List<CommandInfo> Commands { get; set; } = [];
}

File: client/src/app/devices-manager/device.ts
export interface IResultBaseDto<T>  {
  status: boolean;
  message?: string[];
  data?: T;
}

export interface Device {
  id: string;
  identifier: string;
  description: string;
  manufacturer: string;
  url: string;
  commands?: Commands[];
}
export interface Commands {
  operation: string
  description: string
  result: string
  format: string,
  command: Command
}

export interface Command {
  commandName: string
  parameters: Parameter[]
}

export interface Parameter {
  name: string
  description: string
}


